From cacc2eae68ed2003b62945e8e377b81dd968c0e3 Mon Sep 17 00:00:00 2001
From: Cody P Schafer <dev@codyps.com>
Date: Mon, 29 Oct 2018 22:36:58 -0400
Subject: [PATCH 1/3] make: pull out optimization & debug flags

Packagers generally want full control of optimization & debug options.
Provide that by making the current `-g` and `-Os` flags the default
content of `CFLAGS` (any value of `CFLAGS` env will override).
---
 Makefile | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 0538b52..54c992b 100644
--- a/Makefile
+++ b/Makefile
@@ -158,7 +158,8 @@ PAHO_CS_SUB_TARGET = ${blddir}/samples/${PAHO_CS_SUB_NAME}
 #FLAGS_EXE = $(LDFLAGS) -I ${srcdir} -lpthread -L ${blddir}
 #FLAGS_EXES = $(LDFLAGS) -I ${srcdir} ${START_GROUP} -lpthread -lssl -lcrypto ${END_GROUP} -L ${blddir}
 
-CCFLAGS_SO = -g -fPIC $(CFLAGS) -D_GNU_SOURCE -Os -Wall -fvisibility=hidden -I$(blddir_work) -DPAHO_MQTT_EXPORTS=1
+CFLAGS ?= -g -Os
+CCFLAGS_SO = -fPIC $(CFLAGS) -D_GNU_SOURCE -Wall -fvisibility=hidden -I$(blddir_work) -DPAHO_MQTT_EXPORTS=1
 FLAGS_EXE = $(LDFLAGS) -I ${srcdir} ${START_GROUP} -lpthread ${GAI_LIB} ${END_GROUP} -L ${blddir}
 FLAGS_EXES = $(LDFLAGS) -I ${srcdir} ${START_GROUP} -lpthread ${GAI_LIB} -lssl -lcrypto ${END_GROUP} -L ${blddir}
 
-- 
2.26.2

